### PR TITLE
Add Protocol Hints

### PR DESCRIPTION
**Pull request by plypaul**
### Description

This PR adds `ProtocolHint` - a way to hint to the inspections in the IDE that a class is supposed to implement a `Protocol` without explicitly inheriting from a `Protocol`. The explicit inheritance can cause issues with Pydantic classes.

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: dbt_semantic_interfaces/implementations/semantic_manifest.py [hunk 1]
```diff
@@ -2,15 +2,22 @@
...
 
 
-class PydanticSemanticManifest(HashableBaseModel):
+class PydanticSemanticManifest(HashableBaseModel, ProtocolHint[SemanticManifest]):
     """Model holds all the information the SemanticLayer needs to render a query."""
 
+    @override
+    def _implements_protocol(self) -> SemanticManifest:  # noqa: D
+        return self
+
```
Comment (tlento) [inline dbt_semantic_interfaces/implementations/semantic_manifest.py:20]: Oh I see, it's this approach. Yeah this is fine. I wish the toolchain just did this for us, but I can see why it's hard to do in the general case.

**Review by tlento — APPROVED** [review]
I wish the signatures weren't so clumsy but if it works, let's go with it for now. I like your idea of making this a class decorator, it's worth looking into that at some point.

**Comment by plypaul** [comment]
> I like your idea of making this a class decorator, it's worth looking into that at some point.

@tlento I tried a few things to make that happen, but I couldn't get it to work. LMK if you have ideas.

**Comment by tlento** [comment]
@plypaul bummer, I doubt I'd do anything different from what you tried.

**Pull request merged**